### PR TITLE
Remove Galaxy egg code

### PR DESCRIPTION
**Pull request by natefoo**
Do not merge until [galaxy #428](https://github.com/galaxyproject/galaxy/pull/428) is merged (and WIP is removed from this PR). The Pulsar side of the cleaning is incomplete.

### CODE REVIEW

**Comment by jmchilton** [comment]
Can we just merge this now?

**Comment by natefoo** [comment]
No, but maybe if I get some time later today I can finish it.

**Comment by natefoo** [comment]
@jmchilton Is there anything special I should do wrt. updating the Galaxy lib? I could just remove the eggs stuff, but it feels more correct to just copy the relevant things directly from Galaxy so they are the latest versions, but obviously you must have done that with modifications to wrap the eggs imports in the past.

**Comment by natefoo** [comment]
One other thing... would `$GALAXY_HOME` even be necessary for anything going forward if we:

1. Documented how to specify a venv to use for running tools
2. Documented which tools shipped with Galaxy depend on things in that venv - @nsoranzo's work in galaxyproject/galaxy#990 should make this easy

**Comment by jmchilton** [comment]
``GALAXY_HOME`` necessary because metadata depends on models and datatypes.

There is a script for synchronizing - see ``update_galaxy_client.sh`` and ``update_galaxy_utils.sh`` in PULSAR_ROOT.

**Comment by natefoo** [comment]
`update_galaxy_utils.sh` pulls `$GALAXY/lib/galaxy/job/runners/util` into `$PULSAR/pulsar/managers/util`, but the Pulsar copy has modifications that Galaxy does not have. Should Pulsar be the canonical source for this, or Galaxy?

- [$PULSAR/pulsar/managers/util](https://github.com/galaxyproject/pulsar/tree/master/pulsar/managers/util)
- [$GALAXY/lib/galaxy/jobs/runners/util](https://github.com/galaxyproject/galaxy/tree/dev/lib/galaxy/jobs/runners/util)

I can correct it (which will just mean merging them) but just wanted to know. Perhaps a `README` in the downstream directory indicating that changes need to be made upstream and then synchronized would be a good idea.

**Comment by natefoo** [comment]
@jmchilton 

Galaxy seems to be missing things that Pulsar has that came from Planemo?:

- `$PULSAR/galaxy/tools/linters/stdio.py`
- `$PULSAR/galaxy/tools/linters/xml_order.py`
- `$PULSAR/galaxy/tools/linters/xsd_util.py`

Should these be synchronized Planemo -> Galaxy -> Pulsar? Any others like this I need to watch out for?

File: galaxy/tools/deps/resolvers/tool_shed_packages.py [hunk 1]
```diff
@@ -24,9 +24,7 @@ def _find_dep_default( self, name, type='package', **kwds ):
             installed_tool_dependency = self._get_installed_dependency( name, type, version=None, **kwds )
             if installed_tool_dependency:
                 dependency = self._get_set_environment_installed_dependency_script_path( installed_tool_dependency, name )
-                is_galaxy_dep = isinstance(dependency, GalaxyPackageDependency)
-                has_script_dep = is_galaxy_dep and dependency.script and dependency.path
-                if has_script_dep:
+                if dependency.script and dependency.path:
```
Comment (nsoranzo) [inline galaxy/tools/deps/resolvers/tool_shed_packages.py]: Probably this should be reverted and Pulsar version merged to Galaxy, this was last changed in commit e97a95a06d5cc0903125d9e46ab9bf33f7759091 .

File: galaxy/tools/linters/xsd.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+except ImportError:
+    etree = None
+
+
+def lint_via_xsd(tool_xml, lint_ctx):
+    if not lint_ctx.use_schema:
+        raise SkipLint()
+    if etree is None:
+        lint_ctx.error("Requested linting via XML Schema but lxml is unavailable")
+        return
```
Comment (nsoranzo) [inline galaxy/tools/linters/xsd.py]: According to https://github.com/natefoo/galaxy/pull/1, this file should be removed.

**Comment by jmchilton** [comment]
Merged with @nsoranzo two fixes.

File: galaxy/tools/deps/resolvers/tool_shed_packages.py (hunk 1 above)
Comment (natefoo) [inline galaxy/tools/deps/resolvers/tool_shed_packages.py]: As always, good catch, thanks @nsoranzo

**Pull request merged**